examples/OpenGL45/matrix_multiplication/matrix_multiplication_version2.shadertrap

```diff
@@ -211,8 +211,13 @@ layout(std430, binding=1) buffer array1_mat_buf{
   mat4x4 array_mat[MAT_SIZE/KERNEL_SIZE][MAT_SIZE/KERNEL_SIZE];
 };
 void main() {
-    uint row = uint(gl_WorkGroupID[0]), col = uint(gl_WorkGroupID[1]);
-    array_mat[row/KERNEL_SIZE][col/KERNEL_SIZE][col%KERNEL_SIZE][row%KERNEL_SIZE] = data[row][col];
+    uint row = uint(gl_WorkGroupID[0]), col = uint(gl_WorkGroupID[1]);    
+    for(int i = 0; i < KERNEL_SIZE; i++){
+      for(int j = 0; j < KERNEL_SIZE; j++){
+          // Converts the 4x4 array of 4x4 matrices to a 16x16array
+          array_mat[row][col][j][i] = data[row * KERNEL_SIZE + i][col * KERNEL_SIZE + j];
+      }
+    }
```

Comment:
I'm fine with the new way you have expressed this, if you like it better. But I was also OK with the way you had done it before, having each thread take care of a single element.

Reply:
@afd I changed the workgroup number to 16x16 and replaced the nested for loops with:

```array_mat[row/KERNEL_SIZE][col/KERNEL_SIZE][col%KERNEL_SIZE][row%KERNEL_SIZE] = data[row][col];```